charts/seed-controlplane/charts/kube-apiserver/values.yaml

```diff
@@ -88,6 +88,7 @@ apiServerResources:
 maxReplicas: 1
 minReplicas: 1
 targetAverageUtilizationCpu: 80
+memoryMetricForHpaEnabled: false
```

Comment:
Can we have the values organised for example such as:
```yaml
hvpa:
  enabled: false
  metrics:
    memory:
      enabled: false
      targetAverageUtilization: 80
    cpu:
      enabled: true
      targetAverageUtilization: 80
```

Reply:
when hvpa is not enabled, we delpoy HPA instead. The HPA template uses the same `values`. It might not be a good idea to have them under `hvpa` or duplicated for HPA as well. WDYT?

Reply:
Wouldn't be cleaner to have them duplicated for `hvpa` and `hpa`?
You also currently introduce a value called `memoryMetricForHpaEnabled` which is only used in the hvpa template.

Reply:
`memoryMetricForHpaEnabled` works for HVPA since hvpa-controller observes any oomkill that might happen, and applies the VPA recommendations regardless of the vpaWeight provided in the hvpa spec. Since HPA alone does not have that ability, it might still be safer to scale horizontally using memory metric when using HPA alone.
With this change, we can expect some rare oomkills when using hvpa (as mentioned by @amshuman-kr's [comment](https://github.com/gardener/gardener/pull/1968#issuecomment-590205774) with a benefit of reduced costs of seeds.
If we see a stable behaviour, we plan to introduce this change for shooted seeds also, along with possibility of including this in only HPA based scaling.

Reply:
Also, if we see the behaviour is not stable, the plan is to re-introduce memory-based scaling for HPA but with higher utilisation threshold (say, 120%) to still somehow optimise on cost still.

Eitherway is a imperfect solution to a problem which can only be truely solved when HVPA supports the scale sub-resource to [control HPA scaling](https://github.com/gardener/hvpa-controller/issues/7) so that we can scale vertically first and horizontally later.

Reply:
I am only suggesting values restructuring (such as `hvpa.vpa.metrics.memory.enabled`) as at least for me checking values such as `targetAverageUtilizationCpu` or `memoryMetricForHpaEnabled` - I don't get in which template they are used.

Reply:
I understand your concern. i have added another [patch](https://github.com/gardener/gardener/pull/1968/commits/461352f126d3cdd93bd6da35576098965d489d78). Please let me know if this helpful.